### PR TITLE
attach-to-chat 1: components and stories

### PR DESCRIPTION
**Pull request by songgao**
This is the v2 of https://github.com/keybase/client/pull/15320 where we decided to have a new design for the conversation selector.

@cecileboucheron How does this look? Let me know how I should tweak it!

cc @mmaxim @jzila 

image

image

image

depends on https://github.com/keybase/client/pull/15383 and https://github.com/keybase/client/pull/15384

### CODE REVIEW

File: shared/chat/conversation-list/choose-conversation.native.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import type {Props} from './choose-conversation'
+
+export default (props: Props) => (
+  <Kb.Box2 direction="horizontal" centerChildren={true} style={{padding: 32}}>
+    <Kb.Text type="BodySmallError">
+      choose-conversation is not supported on mobile. If you see this outside storybook something is wrong.
```
Comment (chrisnojima) [inline shared/chat/conversation-list/choose-conversation.native.js]: why even have this? just use choose-conversation.desktop (it works on both) and just never route here

File: shared/chat/conversation-list/conversation-list-container.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// @flow
+import {namedConnect} from '../../util/container'
+import ConversationList from './conversation-list'
+
+type OwnProps = {
```
Comment (chrisnojima) [inline shared/chat/conversation-list/conversation-list-container.js]: all these prop types should be exact

File: shared/chat/conversation-list/conversation-list-container.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  onSelect?: () => void,
+}
+
+const mapStateToProps = state => ({})
+const mapDispatchToProps = dispatch => ({})
+const mergeProps = (stateProps, dispatchProps, ownProps) => ({
+  rows: [],
```
Comment (chrisnojima) [inline shared/chat/conversation-list/conversation-list-container.js]: ???

File: shared/chat/inbox/row/big-team-channel/container.js [hunk 4]
```diff
@@ -5,10 +5,10 @@ import * as Chat2Gen from '../../../../actions/chat2-gen'
 import {BigTeamChannel} from '.'
 import {connect, isMobile} from '../../../../util/container'
 
-type OwnProps = {|
+type OwnProps = {
```
Comment (chrisnojima) [inline shared/chat/inbox/row/big-team-channel/container.js]: ???

File: shared/chat/conversation-list/choose-conversation.native.js (hunk 1 above)
Comment (songgao) [reply shared/chat/conversation-list/choose-conversation.native.js]: I was just thinking if someone makes a mistake and uses this on native, this'd make it very clear that it shouldn't happen that way.

File: shared/chat/conversation-list/conversation-list-container.js (hunk 3 above)
Comment (songgao) [reply shared/chat/conversation-list/conversation-list-container.js]: Just a placeholder; Will be populated in a follow-up PRs

File: shared/chat/conversation-list/choose-conversation.native.js (hunk 1 above)
Comment (chrisnojima) [reply shared/chat/conversation-list/choose-conversation.native.js]: but then we need a .js.flow and a .native that does nothing. lets just not have it

File: shared/chat/selectable-big-team-channel-container.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import SelectableBigTeamChannel from './selectable-big-team-channel'
 import * as Types from '../constants/types/chat2'
 import {namedConnect} from '../util/container'
 
-type OwnProps = {|
+export type OwnProps = {|
```
Comment (chrisnojima) [inline shared/chat/selectable-big-team-channel-container.js]: why do we need this

File: shared/chat/selectable-big-team-channel.js [hunk 6]
```diff
@@ -73,29 +83,34 @@ export const rowHeight = isMobile ? 64 : 56
 
 const styles = Styles.styleSheetCreate({
   channelname: Styles.platformStyles({
-    common: {flexBasis: '70%'},
+    common: {
+      flexBasis: '70%',
+      flexShrink: 0.3,
```
Comment (chrisnojima) [inline shared/chat/selectable-big-team-channel.js]: why this change

File: shared/fs/send-attachment-to-chat/index.desktop.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import * as Kb from '../../common-adapters'
+import * as Kbfs from '../common'
+import * as Styles from '../../styles'
+import ChooseConversation from '../../chat/conversation-list/choose-conversation'
+import type {Props} from '.'
+
+const SendAttachmentToChat = (props: Props) => (
```
Comment (chrisnojima) [inline shared/fs/send-attachment-to-chat/index.desktop.js]: if we can have these in one file vs 2 i think we should. there's nothing platform specific about the implementation that requires a different file so I think its better to just have index.js and use Styles.isMobile inside

File: shared/chat/selectable-big-team-channel.js (hunk 6 above)
Comment (songgao) [reply shared/chat/selectable-big-team-channel.js]: flexBasis didn't work well. On desktop, if the team name is too long, then we'd end up having channel name taking 70% space that has a bunch of white space. On mobile if team name is too long, we'd have channel name pushed to far right. Looks like flexShrink doesn't work perfectly either, but I'll play with it and see

Comment (songgao) [reply shared/chat/selectable-big-team-channel.js]: OK I think I've found a way to make it work, by using maxWidth and flexShrink. The result is that channelname only grows if it's overflowing, but only up to 70%. If within 70% channelname can show in full, we just shrink teamname until they fit. If both needs shrinking, then it's 3:7 between teamname and channel name. Here are some screenshots:

image

image

File: shared/fs/send-attachment-to-chat/index.desktop.js (hunk 7 above)
Comment (songgao) [reply shared/fs/send-attachment-to-chat/index.desktop.js]: Sounds good; I'll merge them in the next PR.

**Comment by songgao** [comment]
@chrisnojima I addressed all feedback except merging the platform files (which I'm gonna do in next PR), and the CSS changes for the big team row (where I made more changes). Ready for another look; thanks!

File: shared/react-native/ios/Keybase/Storybook.m [hunk 8]
```diff
@@ -24,7 +24,7 @@ + (BOOL)requiresMainQueueSetup
 - (NSDictionary *)constantsToExport
 {
   // Set this to true to enable storybook mode
-  return @{@"isStorybook": @false};
+  return @{@"isStorybook": @true};
```
Comment (songgao) [inline shared/react-native/ios/Keybase/Storybook.m]: oops (reminder to myself to revert this before merging)

**Comment by songgao** [comment]
bump!
